cpp/daal/src/algorithms/linear_model/linear_model_predict_dense_default_batch_impl.i

```diff
@@ -157,11 +174,29 @@ services::Status PredictKernel<algorithmFPType, defaultDense, cpu>::compute(cons
         }
         algorithmFPType * responseBlock = responseRows.get();
         DAAL_INT * pnumResponses        = (DAAL_INT *)&numResponses;
+        DAAL_INT * pnumRows             = (DAAL_INT *)&numRows;
 
-        /* Calculate predictions */
-        computeBlockOfResponses(&numFeatures, &numRows, dataBlock, &nAllBetas, beta, pnumResponses, responseBlock, model->getInterceptFlag());
+        if (dataTable->getDataLayout() & NumericTableIface::soa)
+        {
+            /* Calculate predictions */
+            computeBlockOfResponsesSOA(startRow, &numFeatures, pnumRows, dataTable, &nAllBetas, beta, pnumResponses, responseBlock,
+                                       model->getInterceptFlag());
+        }
+        else
+        {
+            /* Retrieve data blocks associated with input and resulting tables */
+            ReadRows<algorithmFPType, cpu> dataRows(dataTable, startRow, numRows);
```

Comment:
you can move getting of block to `computeBlockOfResponses` (it will align with computeBlockOfResponsesSOA)